pghoard/rohmu/object_storage/swift.py

```diff
@@ -189,13 +189,16 @@ def delete_key(self, key):
 
     def get_contents_to_file(self, key, filepath_to_store_to, *, progress_callback=None):
         temp_filepath = "{}~".format(filepath_to_store_to)
+        done = False
         with open(temp_filepath, "wb") as fp:
             try:
                 metadata = self.get_contents_to_fileobj(key, fp, progress_callback=progress_callback)
                 os.rename(temp_filepath, filepath_to_store_to)
+                done = True
             finally:
                 with suppress(FileNotFoundError):
-                    os.unlink(filepath_to_store_to)
+                    if not done:
+                        os.unlink(filepath_to_store_to)
```

Comment:
Doesn't look like removing this file would really ever be useful but this function can leak the temp file as it is.